inspirehep/modules/orcid/schema.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+import dojson
+
+orcid_overdo = dojson.Overdo()
+
+
+@orcid_overdo.over('title', 'titles')
+def title_rule(self, key, value):
+    subtitles = []
+    titles = []
+    for val in value:
+        if val.get('subtitle'):
+            subtitles.append(val['subtitle'])
+        if val.get('title'):
+            titles.append(val['title'])
+    return {"title": titles,
+            "subtitle": subtitles}
+
+
+@orcid_overdo.over('journal_title', 'publication_info')
+def publication_rule(self, key, value):
+    publication_info = []
+    for publication in value:
+        publication_info.append(publication.get('journal_title'))
+    return publication_info
+
+
+@orcid_overdo.over('short_description', 'abstracts')
+def abstract_rule(self, key, value):
+    abstracts = []
+    for abstract in value:
+        abstracts.append(abstract.get('value'))
+    return abstracts
+
+
+@orcid_overdo.over('external_id', 'arxiv_eprints|dois')
+@dojson.utils.for_each_value
+def external_id_rule(self, key, value):
+    if key == 'dois':
+        return{
+            'external-id-type': 'doi',
+            'external-id-value': value.get('value'),
+            'external-id-url': 'www.example.com/12344'
+        }
+    if key == 'arxiv_eprints':
+        return {
+            'external-id-type': 'arxiv',
+            'external-id-value': value.get('value'),
+            'external-id-url': 'https://arxiv.org/abs/' + value.get('value').partition(':')[2]
```

Comment:
Well it's weird because the schema is different depending on where you are getting it from 👅 . What I am now using is:

```
@orcid_overdo.over('work_external_identifier', 'arxiv_eprints|dois')
@dojson.utils.for_each_value
def external_id_rule(self, key, value):
    if key == 'dois':
        return{
            'external-identifier-type': 'DOI',
            'external-identifier-id': value.get('value'),
            'external-identifier-url': 'www.example.com/12344'
        }
    if key == 'arxiv_eprints':
        return {
            'external-identifier-type': 'ARXIV',
            'external-identifier-id': value.get('value'),
            'external-identifier-url': 'https://arxiv.org/abs/' + value.get('value').partition(':')[2]
        }
```

and Orcid is accepting it.